pkg/kdump/config-client.js

```diff
@@ -113,21 +124,182 @@ export class ConfigFile {
+            for (const key in this.settings.targets) {
+                const oldTarget = this.settings.targets[key];
+                if (oldTarget.type == "mount") {
+                    delete settings._internal[oldTarget.fsType];
+                } else {
+                    delete settings._internal[key];
+                }
+            }
+
+            if (target.type === "nfs") {
+                this._updateSetting(settings, "nfs", [target.server, target.export].join(":"));
+            } else if (target.type === "ssh") {
+                this._updateSetting(settings, "ssh", target.server);
+                if ("sshkey" in target)
+                    this._updateSetting(settings, "sshkey", target.sshkey);
+            } else if (target.type === "raw") {
+                this._updateSetting(settings, "raw", target.partition);
+            } else if (target.type === "mount") {
+                this._updateSetting(settings, target.fsType, target.partition);
+            }
+
+            /* ssh target needs a flattened vmcore for transport */
+            if ("core_collector" in settings._internal &&
+                settings._internal.core_collector.value.includes("makedumpfile")) {
+                if (target.type === "ssh" && !settings._internal.core_collector.value.includes("-F"))
+                    settings._internal.core_collector.value += " -F";
+                else if (settings._internal.core_collector.value.includes("-F"))
+                    settings._internal.core_collector.value =
+                        settings._internal.core_collector.value
+                                .split(" ")
+                                .filter(e => e != "-F")
+                                .join(" ");
+            }
+        }
+        // compression
+        if (this.settings.compression.enabled != settings.compression.enabled) {
+            if (settings.compression.enabled) {
+                // enable compression
+                if ("core_collector" in settings._internal)
+                    settings._internal.core_collector.value = settings._internal.core_collector.value + " -c";
+                else
+                    settings._internal.core_collector = { value: "makedumpfile -c" };
+            } else {
+                // disable compression
+                if ("core_collector" in this.settings._internal) {
+                    // just remove all "-c" parameters
+                    settings._internal.core_collector.value =
+                        settings._internal.core_collector.value
+                                .split(" ")
+                                .filter((e) => { return (e != "-c") })
+                                .join(" ");
+                } else {
+                    // if we don't have anything on this in the original settings,
+                    // we can get rid of the entry altogether
+                    delete settings._internal.core_collector;
+                }
+            }
```

Comment:
These 16 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17544-20220712-132416-8611a291-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)